ENIGMAsystem/SHELL/Platforms/xlib/LINUXjoystick.cpp

```diff
@@ -124,7 +124,15 @@ namespace enigma
       if (joysticks[i])
         handle_joystick(joysticks[i].get());
   }
-}
+}  // namespace enigma
+
+namespace {
+  string device_name(int js_id) {
```

Comment:
I sort of have a small concern with this, perhaps call it `device_file_name` so it's not confused with the device identifier obtained by `JSIOCGNAME`. It sort of doesn't matter since this is not accessible to the user, it can be changed later if necessary.